wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php

```diff
@@ -1061,4 +1076,82 @@ private static function findOfficeId(array $array) {
     return $goonjOfficeIndex !== FALSE ? $filteredItems[$goonjOfficeIndex] : FALSE;
   }
 
+  /**
+   * This hook is called after a db write on entities.
+   *
+   * @param string $op
+   *   The type of operation being performed.
+   * @param string $objectName
+   *   The name of the object.
+   * @param int $objectId
+   *   The unique identifier for the object.
+   * @param object $objectRef
+   *   The reference to the object.
+   */
+  public static function createActivityForCollectionCamp(string $op, string $objectName, $objectId, &$objectRef) {
+    if ($objectName != 'Eck_Collection_Camp') {
+      return;
+    }
+
+    $newStatus = $objectRef['Collection_Camp_Core_Details.Status'] ?? '';
+
+    if (!$newStatus || !$objectId) {
+      return;
+    }
+
+    $collectionCamp = EckEntity::get('Collection_Camp', FALSE)
+      ->addSelect('Collection_Camp_Core_Details.Status', 'Collection_Camp_Core_Details.Contact_Id', 'title')
+      ->addWhere('id', '=', $objectId)
+      ->execute()->single();
+
+    $currentStatus = $collectionCamp['Collection_Camp_Core_Details.Status'];
+
+    // Check for status change.
+    if ($currentStatus !== $newStatus & $newStatus === 'authorized') {
```

Comment:
`&` and `&&` are two different operators in PHP. First one is bitwise & while the other is logical and

```suggestion
    if ($currentStatus !== $newStatus && $newStatus === 'authorized') {
```

Reply:
Also let's do the negation instead:

```suggestion
    if ($currentStatus === $newStatus || $newStatus !== 'authorized') {
       return;
    }
```